### PR TITLE
[bug-1574]: Fix offline bundle registry references

### PR DESCRIPTION
**Pull request by atye**
# Description

The images.manifest file prior to 1.7.0, used to update the bundle with the private registry when preparing the bundle, contains references with and without docker.io for some images:

```
# grep isilon scripts/images.manifest
dellemc/csi-isilon:v2.11.0
docker.io/dellemc/csi-isilon:v2.11.0
```
If a file contains `docker.io/dellemc/csi-isilon:v2.11.0`, preparing the bundle will replace `dellemc/csi-isilon:v2.11.0` with the private registry: `docker.io/private-registry.com/csi-isilon:v2.11.0`. Then, `docker.io/dellemc/csi-isilon:v2.11.0` is replaced with the private registry but now that string has no matches and the file is left with the faulty registry.

This fix is having all image references specify the registry.

https://github.com/dell/csm-operator/blob/main/bundle/manifests/dell-csm-operator.clusterserviceversion.yaml and https://github.com/dell/csm-operator/blob/main/operatorconfig/driverconfig/common/default.yaml, used to gather image names when building the bundle, are updated to reference all images with the public registry. Since 1.7.0, all of our images now reference quay.io except sdc, otel-collector, and nginx. These are updated to reference docker.io. 

The sample files are updated to reference the registry to be consistent with the images.manifest file.

Made csm-offline-bundle.sh executable.

Adds a new unit test for code coverage improvement for PR check.

Caveats:

- Since https://github.com/dell/csm-operator/blob/main/bundle/manifests/dell-csm-operator.clusterserviceversion.yaml references the latest images, the images.manifest file only has the latest images. The sample files for older releases are not updated with the private registry when preparing the bundle. This behavior has been in place since day one.
- The OCP sample files are not updated with the private registry due to the image names using the redhat registry and SHAs instead of tags. The user will update these sample files manually.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/1574 |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

Built and prepared the bundle verifying that the private registry is correctly put in place in the yaml files that the user will use.
Built and installed the Operator via offline bundle.

### CODE REVIEW

File: samples/authorization/csm_authorization_proxy_server_v1110.yaml [hunk 1]
```diff
@@ -30,12 +30,12 @@ spec:
...
-          proxyService: dellemc/csm-authorization-proxy:v1.11.0
-          tenantService: dellemc/csm-authorization-tenant:v1.11.0
-          roleService: dellemc/csm-authorization-role:v1.11.0
-          storageService: dellemc/csm-authorization-storage:v1.11.0
-          opa: openpolicyagent/opa
-          opaKubeMgmt: openpolicyagent/kube-mgmt:0.11
+          proxyService: docker.io/dellemc/csm-authorization-proxy:v1.11.0
+          tenantService: docker.io/dellemc/csm-authorization-tenant:v1.11.0
+          roleService: docker.io/dellemc/csm-authorization-role:v1.11.0
+          storageService: docker.io/ellemc/csm-authorization-storage:v1.11.0
```
Comment (donatwork) [inline samples/authorization/csm_authorization_proxy_server_v1110.yaml:36]: typo, ellemc -> dellemc

Comment (atye) [reply samples/authorization/csm_authorization_proxy_server_v1110.yaml]: Fixed.

**Review by donatwork — DISMISSED** [review]

**Review by alexemc — DISMISSED** [review]

**Review by donatwork — DISMISSED** [review]

File: pkg/modules/authorization_test.go [hunk 2]
```diff
@@ -225,6 +225,17 @@ func TestAuthInjectDeployment(t *testing.T) {
 			}
 			return true, true, controllerYAML.Deployment, operatorConfig, customResource
 		},
+		"success - greenfield injection mising skip certificate validation env": func(*testing.T) (bool, bool, applyv1.DeploymentApplyConfiguration, utils.OperatorConfig, csmv1.ContainerStorageModule) {
```
[the remark this answers is not in the document]
Comment (atye) [reply pkg/modules/authorization_test.go]: Fixed.

**Review by shaynafinocchiaro — APPROVED** [review]

**Review by jooseppi-luna — COMMENTED** [review]
have you tried installing with the bundle? how have we verified that it works other than looking through it?

**Review by alikdell — APPROVED** [review]

**Review by bharathsreekanth — APPROVED** [review]

**Comment by atye** [comment]
> have you tried installing with the bundle? how have we verified that it works other than looking through it?

Yes, this works. The issue that the customer observed with the RELATED_IMAGES environment variables is resolved and installation is successful, with relevant files updated with the registry.

**Pull request merged**